pkg/util/flavor/flavor.go

```diff
@@ -24,15 +22,12 @@ const (
 	HerokuAgent = "heroku_agent"
 )
 
+//FIXME: This should be a constant defined at build time
```

Comment:
we may want to re-discuss this: I think the main purpose of https://github.com/DataDog/datadog-agent/pull/5769 was precisely to avoid having this defined at build time (but rather in the `main` of each build) since it was apparently confusing for custom builds (I'm not sure what the use cases/examples were though)

Reply:
IIRC the concern was that it was too easy to forget the build flag to set the constant when building an IoT Agent. That meant that one could easily build and IoT Agent that thought it was a full Agent. 

By having a separate main for the IoT Agent the hope was that it was harder to make that mistake.

Reply:
Custom builds should still be of a specific flavor (and be the full agent by default) so I'm not sure why it would be confusing to have flavor-specific behavior defined at build time, but that has nothing to do with this PR so I've removed my comment.